homeassistant/components/permobil/sensor.py

```diff
@@ -0,0 +1,208 @@
+        # Distance that can be traveled with full charge given battery health (km)
+        value_fn=lambda data: data.battery[BATTERY_MAX_DISTANCE_LEFT[0]],
+        key=BATTERY_MAX_DISTANCE_LEFT[0],
+        translation_key="max_distance_left",
+        native_unit_of_measurement=UnitOfLength.KILOMETERS,
+        device_class=SensorDeviceClass.DISTANCE,
+    ),
+    PermobilSensorEntityDescription(
+        # Distance traveled today monotonically increasing, resets every 24h (km)
+        value_fn=lambda data: data.daily_usage[USAGE_DISTANCE[0]],
+        key=USAGE_DISTANCE[0],
+        translation_key="usage_distance",
+        native_unit_of_measurement=UnitOfLength.KILOMETERS,
+        device_class=SensorDeviceClass.DISTANCE,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    PermobilSensorEntityDescription(
+        # Number of adjustments monotonically increasing, resets every 24h
+        value_fn=lambda data: data.daily_usage[USAGE_ADJUSTMENTS[0]],
+        key=USAGE_ADJUSTMENTS[0],
+        translation_key="usage_adjustments",
+        native_unit_of_measurement="adjustments",
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    PermobilSensorEntityDescription(
+        # Largest number of adjustemnts in a single 24h period, never resets
+        value_fn=lambda data: data.records[RECORDS_SEATING[0]],
+        key=RECORDS_SEATING[0],
+        translation_key="record_adjustments",
+        native_unit_of_measurement="adjustments",
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: config_entries.ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Create sensors from a config entry created in the integrations UI."""
+
+    coordinator: MyPermobilCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_entities(
+        PermobilSensor(coordinator=coordinator, description=description)
+        for description in SENSOR_DESCRIPTIONS
+    )
+
+
+class PermobilSensor(CoordinatorEntity[MyPermobilCoordinator], SensorEntity):
+    """Representation of a Sensor.
+
+    This implements the common functions of all sensors.
+    """
+
+    _attr_has_entity_name = True
+    _attr_suggested_display_precision = 0
+    entity_description: PermobilSensorEntityDescription
+    _available = True
+
+    def __init__(
+        self,
+        coordinator: MyPermobilCoordinator,
+        description: PermobilSensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator=coordinator)
+        self.entity_description = description
+        self._attr_unique_id = (
+            f"{coordinator.p_api.email}_{self.entity_description.key}"
+        )
+
+    @property
+    def available(self) -> bool:
+        """Return True if the sensor has value."""
+        try:
+            self.entity_description.value_fn(self.coordinator.data)
```

Comment:
You should key if the key is in the data and not run the method twice....

You should have a function, which returns a boolean (Like the example I provided in my last comment)

Reply:
I dont quite understand, in the [example](https://github.com/isaknyberg/core/blob/e789366839be908e2bbbb940e3e059e01d9f2755/homeassistant/components/lametric/switch.py#L84)  `value_fn` is also run? 
The way it works right now is that if the `value_fn` runs without issue, the value that we are looking for exists in the coordinator dict and then the `available` function returns True. If the value does not exist a keyError is raised by `value_fn` which is caught by the `available` function which then returns False.

Reply:
Is this a feasible alternative?
```
    _available = False

    @property
    def available(self) -> bool:
        """Return True if the sensor has value."""
        return self._available

    @property
    def native_value(self) -> float | None:
        """Return the value of the sensor."""
        try:
            _available = True
            return self.entity_description.value_fn(self.coordinator.data)
        except KeyError: 
            _available = False
            return None
```

Reply:
Oh right, I misread the code. I will amend this. Thank you!

Reply:
I have now implemented the `available_fn`, Should i now remove the try-except clause for the `native_value` since if the value does not exist in the response the sensor will be marked as unavailable immediately?

Reply:
Yes please remove the try/except block